lars-reimann/api-editor
# feat: heatmap mode "usefulness"

Closes #557.

### Summary of Changes

Add a new heat map mode with logarithmic interpolation that show _usefulness_. We define usefulness as

* number of usages for classes and functions
* number of times set to value other than most common one for parameters

### Screenshots (if necessary)

#### Settings

![image](https://user-images.githubusercontent.com/2501322/172064196-f417d8a2-64c6-439b-a327-f0a836530806.png)

#### Heat map

![image](https://user-images.githubusercontent.com/2501322/172064246-fdbd82a1-5b26-4edc-aea5-ba1a4ca77e2d.png)

### Testing Instructions

1. Run this:
```
cd api-editor
./gradlew :backend:run
```
2. Open localhost:4280 in the browser
3. In the settings active the "Usefulness" heatmap mode
4. Check the displayed values
